### DISCUSSION TITLE
Tasmota Smart Meter hardware watchdog reset

### DISCUSSION THREAD

**Discussion by de666** [arendst/Tasmota#20178] [Support request]
### PROBLEM DESCRIPTION
 I compile and configure a wemos D1 mini to read from a modbus RTU power analyzer.
Everything is working good but I see cyclic restart (approx every 10 hours) caused by the hardware watchdog.

This is the smart meter script:
```
>D
>B
->sensor53 r
;->sensor53 d1
>M 1
+1,13,m,0,9600,,15,10,r010400000001,r010400030001,r010400070001,r010400140001,r0104001A0001,0104001D
1,010402SSss@i0:10,Tensione,V,Voltage,1
1,010402SSss@i1:100,Corrente,A,Current,2
1,010402SSss@i2:1,Potenza attiva,W,Power,0
1,010402SSss@i3:1000,Fattore di potenza,,Factor,3
1,010402SSss@i4:100,Frequenza,Hz,Frequency,2
1,010404S32@i5:100,Energia - totale,kWh,Energy,2
#
```

can you give me some help to figure out what's happening?
I'm not able to use the "standard" pins for UART (1/3) since they are bind to the CH340G.
I switch to the alternative serial pins (13/15) but I'm not really sure Tasmota has configured those pin as hardware serial and I don't know how to do it thrugh the scripting (smart meter) interface.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): _____
- [x] Tasmota binary firmware version number used: 13.2.0
  - [ ] Pre-compiled
  - [x] Self-compiled
- [x] Flashing tools used: tasmotizer
- [X] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
12:06:13.622 MQT: stat/tasmota_CF37B1/RESULT = {"NAME":"Generic","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":18}
12:06:13.849 MQT: stat/tasmota_CF37B1/RESULT = {"Module":{"18":"Generic"}}
12:06:14.102 MQT: stat/tasmota_CF37B1/RESULT = {"GPIO0":{"0":"Nessuno"},"GPIO1":{"0":"Nessuno"},"GPIO2":{"320":"Led_i1"},"GPIO3":{"0":"Nessuno"},"GPIO4":{"0":"Nessuno"},"GPIO5":{"0":"Nessuno"},"GPIO9":{"0":"Nessuno"},"GPIO10":{"0":"Nessuno"},"GPIO12":{"0":"Nessuno"},"GPIO13":{"0":"Nessuno"},"GPIO14":{"0":"Nessuno"},"GPIO15":{"0":"Nessuno"},"GPIO16":{"0":"Nessuno"},"GPIO17":{"0":"Nessuno"}}

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
12:07:29.404 CMD: Status 0
12:07:29.412 MQT: stat/tasmota_CF37B1/STATUS = {"Status":{"Module":18,"DeviceName":"DDS669","FriendlyName":["Consumi"],"Topic":"tasmota_CF37B1","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
12:07:29.419 MQT: stat/tasmota_CF37B1/STATUS1 = {"StatusPRM":{"Baudrate":9600,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"","RestartReason":"Hardware Watchdog","Uptime":"0T02:03:52","StartupUTC":"2023-12-06T09:03:37","Sleep":50,"CfgHolder":4617,"BootCount":70,"BCResetTime":"2023-10-28T16:48:42","SaveCount":1657,"SaveAddress":"3FB000"}}
12:07:29.426 MQT: stat/tasmota_CF37B1/STATUS2 = {"StatusFWR":{"Version":"13.2.0(sm)","BuildDateTime":"2023-11-25T11:50:46","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"336/699"}}
12:07:29.432 MQT: stat/tasmota_CF37B1/STATUS3 = {"StatusLOG":{"SerialLog":0,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["xxxx","yyyy"],"TelePeriod":30,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A192800000000","00000080","00006000","00004000","00000000"]}}
12:07:29.443 MQT: stat/tasmota_CF37B1/STATUS4 = {"StatusMEM":{"ProgramSize":577,"Free":1468,"Heap":14,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"1640E0","FlashFrequency":40,"FlashMode":"DOUT","Features":["00000410","0700058A","04080001","00000003","00000000","00000008","00000080","00200000","04000000","00000080"],"Drivers":"1,2,4,9,10,12,38,50,62","Sensors":"1,53"}}
12:07:29.451 MQT: stat/tasmota_CF37B1/STATUS5 = {"StatusNET":{"Hostname":"tasmota-CF37B1-6065","IPAddress":"192.168.1.133","Gateway":"192.168.1.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.1.11","DNSServer2":"208.67.222.222","Mac":"18:FE:34:CF:37:B1","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
12:07:29.457 MQT: stat/tasmota_CF37B1/STATUS6 = {"StatusMQT":{"MqttHost":"mqtt.lan","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_CF37B1","MqttUser":"DVES_USER","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
12:07:29.466 MQT: stat/tasmota_CF37B1/STATUS7 = {"StatusTIM":{"UTC":"2023-12-06T11:07:29","Local":"2023-12-06T12:07:29","StartDST":"2023-03-26T01:00:00","EndDST":"2023-10-29T01:00:00","Timezone":99,"Sunrise":"07:33","Sunset":"16:36"}}
12:07:29.474 MQT: stat/tasmota_CF37B1/STATUS10 = {"StatusSNS":{"Time":"2023-12-06T12:07:29","":{"Voltage":225.8,"Current":0.34,"Power":67,"Factor":0.848,"Frequency":50.00,"Energy":37.38}}}
12:07:29.483 MQT: stat/tasmota_CF37B1/STATUS11 = {"StatusSTS":{"Time":"2023-12-06T12:07:29","Uptime":"0T02:03:52","UptimeSec":7432,"Vcc":3.040,"Heap":14,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"reteDcasa3","BSSId":"C8:7F:54:4E:99:90","Channel":9,"Mode":"11n","RSSI":100,"Signal":-36,"LinkCount":1,"Downtime":"0T00:00:03"}}}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
no hardware watchdog reset

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment]
One typical issue behind stalls (causing the watchdog to bite) is about bad 3.3V power to the ESP8266, and **many** D1 Mini clones have a paltry voltage regulator not suitable for delivering power to attached stuff, like Modbus adapters.

**Reply by de666** [reply]
Good point, I didn't think about it, I will try an external power supply for the 3.3V.
Thankyou very much.
